text/0000-blight-lang.md

```diff
@@ -0,0 +1,159 @@
+- Feature Name: The Blight Programming Language
+- Start Date: 2023-11-03
+- RFC PR: [blight-js/rfcs#0000](https://github.com/blight-js/rfcs/pull/1)
+- Blight Issue:
+  [blight-js/blight#0000](https://github.com/blight-js/blight/issues/1)
+
+# Summary
+
+[summary]: #summary
+
+Javascript today feels very hard to write and this is no doubt due to a number
+of bad decisions and bad features that continues to haunt the web to this day. I
+believe we can do better as a community, if we start from scratch with a better
+language. That being said, I strongly believe that Javascript as a language in
+browsers is going nowhere and will stay the way it is forever.
+
+Without browser support, it's impossible for us to get a good language in
+browsers through traditional means, so until we get a big enough community to
+back us entering browsers by default, we will happily transpile all Blight code
+to Javascript that can run on the server or in the browser.
+
+TLDR; Blight is Typescript 2, inspired by Rust syntax and semantics.
+
+# Motivation
+
+[motivation]: #motivation
+
+We are doing this because we think that with the current landscape of browsers,
+good language features are easy to come by, especially with inspiration from the
+sources. We believe that a language that is designed around scripting with some
+safety checks involved would be amazing and we are committed to making it
+happen.
+
+To be very clear, we do not plan on having any ecosystem to begin. An attack
+strategy is being devised to get around that. We do want to support some sort of
+interface with JavaScript and we plan to do that in a future proposal.
+
+In the best timeline, once we get the compiler up and running, we are able to
+create a runtime pretty fast and kick off a hackathon to get an ecosystem
+running. If we get some key members of the JS community together, we may be able
+to get most of the core necessary modules done pretty fast.
```

Comment:
So are there no plans for backward compatibility with JS? I.e., I won't be able to import and run JS files inside of Blight? It seems like I'd be able to, so maybe highlight that.

Reply:
Ah nvm, just read one of the non-goals. So no library support at all, but there is still an interface? Won't having an interface with JS + transpiling to JS imply that I can use JS libraries inside of Blight? Or am I misunderstanding something?

Reply:
No plans for backwards compatibility, that is correct. I'm hoping to kick-start the development of most of the "necessary" libraries/frameworks by just asking the authors (or some of my friends) directly to work on it.

Things that would be great to have early that I might be able to knock out:
- Some sort of reactive web framework
- Some sort of browser automation library

Reply:
And yet there will be transpilation and an interface? Wouldn't this imply support? Like couldn't I just do 
```
unsafeJS {
  import x from "wherever"
}
```

Reply:
`unsafe` blocks might be able to create an interface. There is mixed reception to unsafe blocks in the team.